### ISSUE TITLE
Remove cipher/aead legacy dependencies from PSA test suites

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#8415]
In #8119, a few legacy dependencies in PSA test suites were replaced with PSA dependencies.

However it looks like a few remain:
```
% grep 'depends_on' tests/suites/test_suite_psa* | grep -Eo '!?MBEDTLS_[^: ]*' | grep -v MBEDTLS_PSA_ | sort -u
!MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH
MBEDTLS_ASN1_WRITE_C
MBEDTLS_CIPHER_C
MBEDTLS_ECP_LIGHT
MBEDTLS_ECP_RESTARTABLE
MBEDTLS_ENTROPY_NV_SEED
MBEDTLS_GCM_C
MBEDTLS_MD_C
MBEDTLS_PEM_PARSE_C
MBEDTLS_PK_C
MBEDTLS_PK_PARSE_C
MBEDTLS_RSA_C
```

This task is to eliminate `CIPHER_C` and `GCM_C` from that list, in order to avoid further surprises or test gaps in future work. (`!MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH` seems justified and can be kept.)

From a quick look:
- ~There's only one occurrence of `GCM_C`, in `test_suite_psa_crypto.data`, and it looks like the test does not use GCM, so it probably should be changed to `ALG_CTR` actually.~ fixed by #8303
- There are a few occurrences of `CIPHER_C` in `test_suite_psa_crypto_metadata.data` and it looks they're all redundant and can be simply removed.

**Depends on:** #8119
